leshan-core/src/main/java/org/eclipse/leshan/core/model/DdfList2JsonGenerator.java

```diff
@@ -0,0 +1,115 @@
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class DdfList2JsonGenerator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DdfList2JsonGenerator.class);
+
+    private final DocumentBuilderFactory factory;
+
+    public DdfList2JsonGenerator() {
+        factory = DocumentBuilderFactory.newInstance();
+    }
+
+    private void processDdfList(String url, String ddfFilesPath) throws IOException {
+
+        LOG.debug("Processing DDF list file {}", url);
+
+        List<String> ddfUrls = new ArrayList<>();
+
+        try {
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document document = builder.parse(url);
+
+            NodeList items = document.getDocumentElement().getElementsByTagName("Item");
+            for (int i = 0; i < items.getLength(); i++) {
+                Node item = items.item(i);
+                Node ddf = ((Element) item).getElementsByTagName("DDF").item(0);
+                ddfUrls.add(ddf.getTextContent());
+            }
+        } catch (SAXException | ParserConfigurationException e) {
+            throw new IOException(e);
+        }
+
+        for (String ddfUrl : ddfUrls) {
+
+            URL parsedUrl;
+            try {
+                parsedUrl = new URL(ddfUrl);
+            } catch (MalformedURLException e) {
+                LOG.error("Skipping malformed URL {}", ddfUrl);
+                continue;
+            }
+
+            String filename = parsedUrl.getPath();
+            filename = filename.substring(filename.lastIndexOf("/"));
+
+            Path outPath = Paths.get(ddfFilesPath, filename);
+
+            LOG.debug("Downloading DDF file {} to {}", ddfUrl, outPath);
+
+            URLConnection conn = parsedUrl.openConnection();
+            // The default Java User-Agent gets 403 Forbidden from OMA website
+            conn.setRequestProperty("User-Agent", "Leshan " + getClass().getSimpleName());
+
+            try (InputStream in = conn.getInputStream()) {
+                Files.copy(in, outPath);
+            }
+        }
+    }
+
+    public static void main(String[] args) throws IOException {
+        // default values
+        String ddfListUrl = "https://raw.githubusercontent.com/OpenMobileAlliance/openmobilealliance.github.io/master/OMNA/LwM2M/DDF.xml";
+        String ddfFilesPath = Ddf2JsonGenerator.DEFAULT_DDF_FILES_PATH;
+
+        // use arguments if they exist
+        if (args.length >= 1)
+            ddfFilesPath = args[0]; // the path to a DDF file or a folder which contains DDF files.
+        if (args.length >= 2)
+            ddfListUrl = args[1]; // the path of the DDF list file.
+
+        new DdfList2JsonGenerator().processDdfList(ddfListUrl, ddfFilesPath);
+
+        LOG.debug("DDF list {} processed to {}, proceeding with JSON generation", ddfListUrl, ddfFilesPath);
+
+        Ddf2JsonGenerator.main(new String[] { ddfFilesPath });
```

Comment:
We should add a third parameter for the path of the json output path and use :
```java
        // generate object spec file
        Ddf2JsonGenerator ddfJsonGenerator = new Ddf2JsonGenerator();
        try (FileOutputStream fileOutputStream = new FileOutputStream(outputPath)) {
            ddfJsonGenerator.generate(new File(ddfFilesPath), fileOutputStream);
        }
```
instead of : 
```java
 Ddf2JsonGenerator.main(new String[] { ddfFilesPath });
```